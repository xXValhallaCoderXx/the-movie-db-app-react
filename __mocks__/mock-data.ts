import {
  ISelectedMovie,
  IMovieDetail,
  ICast,
  IReviews
} from "../src/shared/types";

export const castData: ICast[] = [
  {
    cast_id: 1,
    character: "Pikachu",
    name: "Pikachuuuuu",
    profile_path: "/img.jpg"
  }
];

export const reviewData: IReviews[] = [
  {
    author: "Ash Catchem",
    content: "This is the review",
    url: "www.urlreview.com"
  }
];

export const searchMovieResponse: IMovieDetail[] = [
  {
    id: "1",
    backdrop_path: "/img.jpg",
    budget: 2000,
    homepage: "www.marvel.com",
    overview: "This is the overview",
    imdb_id: "1234",
    poster_path: "/img.jpg",
    revenue: 2000000,
    release_date: "",
    runtime: 143,
    status: "released",
    tagline: "To infinity and beyond",
    title: "Avengers",
    genres: [
      {
        id: 28,
        name: "Action"
      }
    ]
  }
];

export const mockPaginationData = (data): IMovieDetail[] => {
  const mockData: IMovieDetail[] = [];
  for (let i = 0; i < 61; i++) {
    mockData.push(data);
  }
  return mockData;
};

export const invalidSearchResponse: any[] = [
  {
    id: "1",
    backdrop_path: "/img.jpg",
    budget: 2000,
    homepage: "www.marvel.com",
    overview: "This is the overview",
    imdb_id: "1234",
    poster_path: "/img.jpg",
    revenue: 2000000,
    release_date: "",
    runtime: 143,
    status: "released",
    tagline: "To infinity and beyond",
    genres: [
      {
        id: 28,
        name: "Action"
      }
    ]
  }
];

export const searchResults = [
  {
    id: 603,
    title: "The Matrix"
  }
];

export const selectedMovie: ISelectedMovie = {
  cast: [
    {
      cast_id: 34,
      character: "Thomas A. Anderson / Neo",
      name: "Keanu Reeves",
      profile_path: "/bOlYWhVuOiU6azC4Bw6zlXZ5QTC.jpg"
    }
  ],
  info: {
    id: "1",
    backdrop_path: "/img.jpg",
    budget: 2000,
    homepage: "www.marvel.com",
    overview: "This is the overview",
    imdb_id: "1234",
    poster_path: "/img.jpg",
    revenue: 2000000,
    release_date: "",
    runtime: 143,
    status: "released",
    tagline: "To infinity and beyond",
    title: "Avengers",
    genres: [
      {
        id: 28,
        name: "Action"
      }
    ]
  },
  reviews: [
    {
      author: "GeekMasher",
      content:
        "The Martix is a great example of a movie that will live for ever or a very log time. The story and concept are out of this world. Keanu Reeves plays his role with utter brilliance, the cast was very well put together and the graphics are still to this day amazing. All in all one of the best movies of all time.",
      url: "https://www.themoviedb.org/review/51429fb419c29552f00f2f7b"
    }
  ],
  similar: [
    {
      id: 605,
      title: "The Matrix Revolutions",
      poster_path: "/sKogjhfs5q3azmpW7DFKKAeLEG8.jpg"
    }
  ]
};
