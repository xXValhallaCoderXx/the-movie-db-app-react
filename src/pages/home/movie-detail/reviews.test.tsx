import React from "react";
import {reviewData} from "../../../../__mocks__/mock-data";
import {shallow} from "enzyme";
import Reviews from "./reviews";

describe("Movie Details - Reviews", () => {
  it("should render correctly without data", () => {
    const component = shallow(<Reviews reviews={[]} />);
    expect(component.find("#no-review").text()).toEqual("No Review Data");
    expect(component).toMatchSnapshot();
  });
  it("should render correctly with data", () => {
    const component = shallow(<Reviews reviews={reviewData} />);
    expect(component.find("#review-author").text()).toEqual(
      reviewData[0].author
    );
    expect(component.find("#review-content").text()).toEqual(
      reviewData[0].content
    );
    expect(component).toMatchSnapshot();
  });
});
