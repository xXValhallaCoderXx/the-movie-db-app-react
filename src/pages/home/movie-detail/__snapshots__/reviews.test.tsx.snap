// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details - Reviews should render correctly with data 1`] = `
<div
  className="flex flex-wrap justify-center mt-5"
>
  <div
    className="mt-10 md:mt-0 w-full md:w-1/3 bg-white rounded text-center p-3 m-5"
    key="0"
  >
    <p
      className="text-center font-bold mt-4"
      id="review-author"
    >
      Ash Catchem
    </p>
    <p
      className="text-sm"
      id="review-content"
    >
      This is the review
    </p>
    <a
      className="mt-2 cursor-pointer hover:underline text-wrap font-semibold hover:font-bold text-tmd-green"
      href="www.urlreview.com"
      target="_blank"
    >
      Read More
    </a>
  </div>
</div>
`;

exports[`Movie Details - Reviews should render correctly without data 1`] = `
<div
  className="text-center font-bold text-xl"
  id="no-review"
>
  No Review Data
</div>
`;
